### PR TITLE
Include particle plots and comparisons [com1;ana5;out3;in1;ana3;in3;out3]

### PR DESCRIPTION
**Pull request by fso42**
Stuff done by @oscar-dick

Updates from AvaFrame:

    removed folder Tools and incorporated plot functions into out3Plot/outParticlesAnalysis.py
    removed all calls to Avanodes
    renamed run script from runFrictionTest.py to runScripts/runParticleAnalysisPlots.py
    restructure run script to perform analysis and plots for one sim at a time
    call already existing functions instead of creating them again to avoid duplicated code
    add option to run com1DFA sim from run script
    add outParticlesAnalysisCfg.ini with override sections for com1DFA, ana3AIMEC and distanceTimeAna

-> still required: pytests for new functions like envelope computation

See https://github.com/avaframe/AvaFrame/issues/876 for documentation related info

### CODE REVIEW

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 1]
```diff
@@ -8,13 +8,64 @@
 
 # Local imports
 import avaframe.ana3AIMEC.aimecTools as aT
+from avaframe.ana3AIMEC import aimecTools
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:11]: These 2 are redundant, right?

File: avaframe/ana5Utils/DFAPathGenerationCfg.ini [hunk 2]
```diff
@@ -43,3 +43,6 @@ dsMin = 20
 
 # True to get the path from particles, False to get the path from fields (FT and FM)
 pathFromPart = True
+
+# parameters for scipy.interpolate splprep
+uInterval = 1000
```
Comment (fso42) [inline avaframe/ana5Utils/DFAPathGenerationCfg.ini:48]: Didn't we say to get rid of this? And it is not used anymore, right?

File: avaframe/ana5Utils/distanceTimeAnalysisCfg.ini [hunk 3]
```diff
@@ -51,6 +51,9 @@ maxOrMean = mean
 # options: projected, parallel
 sType = projected
 
+# origin of s coordinate - if True set to start of thalweg, if False use with respect to startOfRunoutAreaAngle point
```
Comment (fso42) [inline avaframe/ana5Utils/distanceTimeAnalysisCfg.ini:54]: Maybe add: this is relevant for plotting only?

File: avaframe/out3Plot/outCom1DFA.py [hunk 4]
```diff
@@ -123,13 +160,18 @@ def plotTrackParticleAcceleration(outDirData,trackedPartProp, cfg):
...
+    for idx in particlesList[0]['ID']:
+        uAcc[:,idx] = np.asarray([p['uAcc'][idx] for p in particlesList])
+        ax1.plot(Tsave, uAcc[:,idx])
     ax1.set_xlabel('time step [s]')
     ax1.set_ylabel('acceleration [ms-2]')
 
     pathDict = {}
     pathDict['pathResult'] = outDirData
-    outFileName = 'trackedParticles_acceleration'
+    outFileName = 'Allparticles_acceleration_%s' % cuSimName
```
Comment (fso42) [inline avaframe/out3Plot/outCom1DFA.py:174]:
```suggestion
    outFileName = 'allParticles_acceleration_%s' % cuSimName
```

File: docs/moduleAna5Utils.rst [hunk 5]
```diff
@@ -240,3 +240,14 @@ the default one (``fitOption``= 0) minimises the distance between the parabolic
 profile. The second option (``fitOption``= 1) matches the end slope of the parabola to the profile.
 This parabolic fit determines the split point location. It is the first point for which the slope is
 lower than the ``slopeSplitPoint`` angle. This point is then projected on the avalanche path profile.
+
+Resampling
+========================================
+
+To automatically generate the avalanche thalweg, the center of the avalanche mass is calculated at every time step of the simulation computation and a curve linking these points is interpolated using a spline with polynomials of order k. However, depending on the velocity of the flow between two steps, the points can be more or less close to each other.
```
Comment (fso42) [inline docs/moduleAna5Utils.rst:247]:
```suggestion
To automatically generate the avalanche thalweg, the center of the avalanche mass is calculated at every time 
step of the simulation computation, and a curve linking these points is interpolated using a spline with 
polynomials of order k. However, depending on the flow velocity between two steps, the points can be 
more or less close to each other. 
```

File: docs/moduleAna5Utils.rst [hunk 6]
```diff
@@ -240,3 +240,14 @@ the default one (``fitOption``= 0) minimises the distance between the parabolic
 profile. The second option (``fitOption``= 1) matches the end slope of the parabola to the profile.
 This parabolic fit determines the split point location. It is the first point for which the slope is
 lower than the ``slopeSplitPoint`` angle. This point is then projected on the avalanche path profile.
+
+Resampling
+========================================
+
+To automatically generate the avalanche thalweg, the center of the avalanche mass is calculated at every time step of the simulation computation and a curve linking these points is interpolated using a spline with polynomials of order k. However, depending on the velocity of the flow between two steps, the points can be more or less close to each other. 
+
+Especially in the release and runout area, the extracted centers of mass are close to each other, which results sometimes in grid lines drawn perpendicularly to the thalweg overlapping over the width of the domain. In order to reduce these overlaps, we can resample the mass averaged path, as long as the distance between two points remains lower or equal to the spatial resolution of the topography DEM.
```
Comment (fso42) [inline docs/moduleAna5Utils.rst:249]:
```suggestion
Especially in the release and runout area, the extracted centers of mass are close to each other, which sometimes results
 in grid lines drawn perpendicularly to the thalweg overlapping over the width of the domain. In order to reduce 
 these overlaps, we can resample the mass averaged path, as long as the distance between two points remains 
 lower or equal to the spatial resolution of the topography DEM.
```

File: avaframe/ana5Utils/DFAPathGenerationCfg.ini (hunk 2 above)
Comment (awirb) [reply avaframe/ana5Utils/DFAPathGenerationCfg.ini:48]: is not used anymore

File: avaframe/out3Plot/outCom1DFA.py [hunk 7]
```diff
@@ -114,7 +116,42 @@ def plotTrackParticleAcceleration(outDirData,trackedPartProp, cfg):
...
+    fig.suptitle('Tracked particles acceleration')
+    ax1 = plt.subplot(111)
+
+    ax1.plot(trackedPartProp['t'], trackedPartProp['uAcc'])
+    ax1.set_xlabel('time step [s]')
+    ax1.set_ylabel('acceleration [ms-2]')
+
+    pathDict = {}
+    pathDict['pathResult'] = outDirData
+    outFileName = 'trackedParticles_acceleration_ %s' % cuSimName
```
Comment (fso42) [inline avaframe/out3Plot/outCom1DFA.py:136]:
```suggestion
    outFileName = 'trackedParticlesAcceleration_%s' % cuSimName
```

File: avaframe/out3Plot/outCom1DFA.py [hunk 8]
```diff
@@ -123,13 +160,18 @@ def plotTrackParticleAcceleration(outDirData,trackedPartProp, cfg):
...
+    for idx in particlesList[0]['ID']:
+        uAcc[:,idx] = np.asarray([p['uAcc'][idx] for p in particlesList])
+        ax1.plot(Tsave, uAcc[:,idx])
     ax1.set_xlabel('time step [s]')
     ax1.set_ylabel('acceleration [ms-2]')
 
     pathDict = {}
     pathDict['pathResult'] = outDirData
-    outFileName = 'trackedParticles_acceleration'
+    outFileName = 'allparticles_acceleration_%s' % cuSimName
```
Comment (fso42) [inline avaframe/out3Plot/outCom1DFA.py:174]:
```suggestion
    outFileName = 'allParticlesAcceleration_%s' % cuSimName
```

**Pull request merged**